### PR TITLE
Symbol::bind and Executor forward/backward

### PR DESCRIPTION
**Pull request by javelinjs**
* Most functions of Symbol finished
* Executor finished, and tested
* Wrote some functions for Model, but not finished

@yanqingmen I'm quite sure that DataBatch.data (and DataBatch.label) is IndexedSeq[NDArray] (instead of NDArray, see executor.py::_load_data and test_io.py::test_Cifar10Rec, test_NDArrayIter). However, get_data() in PrefetchingIter returns array of NDArray while get_data() in MXDataIter returns NDArray. For clarity, I made MXDataIter::getData() also return IndexedSeq[NDArray]. Please pay attention to these modifications.

### CODE REVIEW

**Comment by terrytangyuan** [comment]
Good work!

**Comment by yanqingmen** [comment]
greate! 
I'll take care of these modifications

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/Executor.scala [hunk 1]
```diff
@@ -19,12 +21,11 @@ object Executor {
    * @throws IllegalArgumentException
    *         If there are two many splits such that some slice can be empty.
    */
-  private def splitInputSlice[@specialized(Int, Float, Double) V]
-                              (batchSize: Int, workLoadList: Array[V])
-                              (implicit num: Numeric[V]): Array[(Int, Int)] = {
-    val totalWorkLoad = workLoadList.sum.asInstanceOf[Float]
+  private[mxnet] def splitInputSlice(batchSize: Int,
```
Comment (terrytangyuan) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/Executor.scala]: So we only accept Float for `workLoadList` now? Is this consistent with other places?

**Comment by terrytangyuan** [comment]
Otherwise, looks good to me. Great work!

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/Executor.scala (hunk 1 above)
Comment (javelinjs) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/Executor.scala]: Yes. No worth to support integer which will introduce overheads

**Comment by yanqingmen** [comment]
LGTM

**Pull request merged**